app/src/main/java/com/grarak/kerneladiutor/fragments/other/SettingsFragment.java

```diff
@@ -324,7 +324,7 @@ private void dumpsysfs() {
         // loop through each array in the constants file. These contain all the other arrays.
         // have to do this once for the 1d arrays and again for the 2 arrays
         try {
-            File sysfsdump = new File(MainActivity.context.getFilesDir(), "sysfsdump.txt");
+            File sysfsdump = File.createTempFile("sysfsdump", ".txt", MainActivity.context.getExternalMediaDirs()[0]);
```

Comment:
http://www.tutorialspoint.com/java/io/file_createtempfile_directory.htm

Unless you call deleteOnExit() this will not do what you intended.

Android has it's own temporary file APIs  getCacheDir()